### ISSUE TITLE
X server crashes when attempting to Auto-Type (2.7.0 AppImage on Debian/Gnome)

### ISSUE THREAD

**Issue by jkraemer** [keepassxreboot/keepassxc#7797]
## Overview

I'm trying out the 2.7.0 AppImage on Debian Buster with Gnome. When using auto-type in a browser (I tried Firefox and Chromium), the X session crashes and I have to log in again. If there are multiple possible entries in the database, the small window to select an entry appears, however on selecting an entry, the crash happens.

Version 2.6.2 from buster-backports works fine. Unfortunately I have a database here that apparently was created on 2.7.0 and does not open in previous versions (is that something I should report separately, or is this by design?).

## Steps to Reproduce

1. Log into a Gnome sesion on Debian 10 
2. Launch the keepassxc 2.7.0 AppImage, open a database
3. Launch firefox, navigate to  form that has a matching entry in the KeepassXC database
4. Hit Alt-\ on to trigger Auto-Type

## Expected Behavior

username / password should be filled in

## Actual Behavior

X session crashes, I end up at the login screen.

## Context

I found this in `/var/log/messages`:

~~~
Apr  5 17:02:09 x270 org.gnome.Shell.desktop[18264]: (EE)
Apr  5 17:02:09 x270 org.gnome.Shell.desktop[18264]: (EE) Backtrace:
Apr  5 17:02:09 x270 org.gnome.Shell.desktop[18264]: (EE) 0: /usr/bin/Xwayland (OsLookupColor+0x139) [0x55641ba05c59]
Apr  5 17:02:09 x270 org.gnome.Shell.desktop[18264]: (EE) 1: /lib/x86_64-linux-gnu/libpthread.so.0 (funlockfile+0x50) [0x7ff14164c77f]
Apr  5 17:02:09 x270 org.gnome.Shell.desktop[18264]: (EE) 2: /usr/bin/Xwayland (XkbCompileKeymapFromString+0x228d) [0x55641b98edad]
Apr  5 17:02:09 x270 org.gnome.Shell.desktop[18264]: (EE) 3: /usr/bin/Xwayland (XkbCompileKeymapFromString+0x9bb7) [0x55641b99ddf7]
Apr  5 17:02:09 x270 org.gnome.Shell.desktop[18264]: (EE) 4: /usr/bin/Xwayland (SendErrorToClient+0x35e) [0x55641b9cfbfe]
Apr  5 17:02:09 x270 org.gnome.Shell.desktop[18264]: (EE) 5: /usr/bin/Xwayland (InitFonts+0x3b6) [0x55641b9d3bc6]
Apr  5 17:02:09 x270 org.gnome.Shell.desktop[18264]: (EE) 6: /lib/x86_64-linux-gnu/libc.so.6 (__libc_start_main+0xeb) [0x7ff14149d09b]
Apr  5 17:02:09 x270 org.gnome.Shell.desktop[18264]: (EE) 7: /usr/bin/Xwayland (_start+0x2a) [0x55641b8a423a]
Apr  5 17:02:09 x270 org.gnome.Shell.desktop[18264]: (EE)
Apr  5 17:02:09 x270 org.gnome.Shell.desktop[18264]: (EE) Segmentation fault at address 0x68
Apr  5 17:02:09 x270 org.gnome.Shell.desktop[18264]: (EE)
Apr  5 17:02:09 x270 org.gnome.Shell.desktop[18264]: Fatal server error:
Apr  5 17:02:09 x270 org.gnome.Shell.desktop[18264]: (EE) Caught signal 11 (Segmentation fault). Server aborting
Apr  5 17:02:09 x270 org.gnome.Shell.desktop[18264]: (EE)
~~~

KeePassXC - Version 2.7.0
Revision: d7a9ef4
Distribution: AppImage

Qt 5.15.2
Debugging mode is disabled.

Operating system: Debian GNU/Linux 10 (buster)
CPU architecture: x86_64
Kernel: linux 4.19.0-19-amd64

Enabled extensions:
- Auto-Type
- Browser Integration
- SSH Agent
- KeeShare
- YubiKey
- Secret Service Integration

Cryptographic libraries:
- Botan 2.12.1

Desktop Env: Gnome
Windowing System: Wayland ( I guess from the log message?)

**Comment by droidmonkey** [comment]
If it is the x server crashing then you will need to report that to them. 2.7.0 defaults to kdbx 4.1 which is compatible with 2.6.6 only.

**Comment by hifi** [comment]
This looks like you're running Wayland and both KeePassXC and Firefox are running on top of Xwayland. Wayland is a constantly moving target and the versions in Debian buster are fairly old and running X applications on top of Xwayland is even more convoluted.

I'm pretty sure Xwayland - at least the version you are on - is buggy and the Auto-Type changes make it worse on 2.7 as it uses more advance techniques. 

If you don't specifically need/want Wayland you're probably better off using regular old Xorg for compatibility. I don't think we can or should do anything on our side, unfortunately.

**Comment by jkraemer** [comment]
OK, thanks for the background info. I guess I'll stick to 2.6.x and try again after upgrading Debian then.

**Comment by sph3rex** [comment]
This is not specific to XWayland, because I get the same on OpenSuse 15.4 with xorg-x11-server-1.20.3. So the only way to "fix" this is either downgrading to 2.6.6 or wait for some time until xorg fixes this (if they will ever do that)
